### ISSUE TITLE
Trying to understand dir and file mode

### ISSUE THREAD

**Issue by IchordeDionysos** [GoogleCloudPlatform/gcsfuse#2239]
**Describe the issue**
While trying to get gcsfuse to run on Cloud Run now that https://github.com/GoogleCloudPlatform/gcsfuse/issues/1016 is fixed, I'm a bit confused on the file/dir modes that are logged by gcsfuse.

First of all, they don't seem to log what is passed by `--dir-mode` and `--file-mode`.

However, the logged file mode doesn't make sense, `438`. How can the other permissions be 8 (isn't the maximum possible number 7?).

How can if go from the `DirMode` and `FileMode` mound command flags go back to what has been used for `--dir-mode` and `--file-mode`?

Thanks for any insights :)

**To Collect more Debug logs**

```
Cloud Storage FUSE will be mounted to an internal path not accessible from customer containers

severity=INFO message="Start gcsfuse/2.1.0 (Go version go1.22.3) for app \"serverless\" using mount point: /var/lib/volumes/gcs/atlantis_volume\n"

severity=INFO message="GCSFuse mount command flags: {\"AppName\":\"serverless\",\"Foreground\":true,\"ConfigFile\":\"\",\"MountOptions\":{\"allow_other\":\"\"},\"DirMode\":511,\"FileMode\":438,\"Uid\":-1,\"Gid\":-1,\"ImplicitDirs\":true,\"OnlyDir\":\"\",\"RenameDirLimit\":0,\"IgnoreInterrupts\":true,\"CustomEndpoint\":null,\"BillingProject\":\"\",\"KeyFile\":\"\",\"TokenUrl\":\"\",\"ReuseTokenFromUrl\":true,\"EgressBandwidthLimitBytesPerSecond\":-1,\"OpRateLimitHz\":-1,\"SequentialReadSizeMb\":200,\"AnonymousAccess\":false,\"MaxRetrySleep\":30000000000,\"StatCacheCapacity\":20460,\"StatCacheTTL\":60000000000,\"TypeCacheTTL\":60000000000,\"HttpClientTimeout\":0,\"MaxRetryDuration\":-1000000000,\"RetryMultiplier\":2,\"LocalFileCache\":false,\"TempDir\":\"\",\"ClientProtocol\":\"http1\",\"MaxConnsPerHost\":0,\"MaxIdleConnsPerHost\":100,\"EnableNonexistentTypeCache\":false,\"StackdriverExportInterval\":0,\"OtelCollectorAddress\":\"\",\"LogFile\":\"/mnt/logging/gcs/system\",\"LogFormat\":\"text\",\"ExperimentalEnableJsonRead\":false,\"DebugFuseErrors\":true,\"DebugFuse\":false,\"DebugFS\":false,\"DebugGCS\":false,\"DebugHTTP\":false,\"DebugInvariants\":false,\"DebugMutex\":false}"

severity=INFO message="GCSFuse mount config flags: {\"CreateEmptyFile\":false,\"Severity\":\"INFO\",\"Format\":\"text\",\"FilePath\":\"/mnt/logging/gcs/system\",\"LogRotateConfig\":{\"MaxFileSizeMB\":512,\"BackupFileCount\":10,\"Compress\":true},\"MaxSizeMB\":-1,\"CacheFileForRangeRead\":false,\"CacheDir\":\"\",\"TtlInSeconds\":-9223372036854775808,\"TypeCacheMaxSizeMB\":4,\"StatCacheMaxSizeMB\":-9223372036854775808,\"EnableEmptyManagedFolders\":false,\"ConnPoolSize\":1,\"AnonymousAccess\":false,\"EnableHNS\":false,\"IgnoreInterrupts\":true}"
```

**System (please complete the following information):**
 - OS: Alpine 3.19.0
 - Platform Cloud Run
 - Version 2.1.0

**Additional context**

**Comment by raj-prince** [comment]
Thanks @IchordeDionysos for reporting this!!

Yes, this is a bug in the logging of fileMode or dirMode. E.g. `-rwxr-xr-x` (`0111101101`) is converted to 493 (decimal base).

We will fix it.

**Comment by IchordeDionysos** [comment]
Ahh got it, so converting from decimal to octal should do the trick 👌🏽

**Comment by kislaykishore** [comment]
This change has been released in v2.4.1.
